### PR TITLE
Validator rules to check if target suppports federation or inheritance

### PR DESCRIPTION
**Pull request by cmnrd**
Fixes https://github.com/lf-lang/lingua-franca/issues/1652.

This also updates the checkReactor rule in order to mitigate potential exception and avoid weird control flow (i.e. returning from a nested if).

### CODE REVIEW

File: org.lflang/src/org/lflang/validation/LFValidator.java [hunk 1]
```diff
@@ -852,29 +852,22 @@ public void checkReaction(Reaction reaction) {
     // FIXME: improve error message.
     }
 
-    @Check(CheckType.FAST)
-    public void checkReactor(Reactor reactor) throws IOException {
```
Comment (edwardalee) [inline org.lflang/src/org/lflang/validation/LFValidator.java:856]: Why remove this check?

**Review by edwardalee — CHANGES_REQUESTED** [review]

File: org.lflang/src/org/lflang/validation/LFValidator.java (hunk 1 above)
Comment (lhstrh) [reply org.lflang/src/org/lflang/validation/LFValidator.java:856]: I don't think it is removed. See line 900.

File: org.lflang.tests/src/org/lflang/tests/compiler/LinguaFrancaValidationTest.java [hunk 2]
```diff
@@ -812,6 +812,41 @@ public void testPreambleVisibility() throws Exception {
...
+    @Test
+    public void testInheritanceSupport() throws Exception {
+        for (Target target : Target.values()) {
+            Model model = parseWithoutError("""
+                    target %s
+                    reactor A{}
+                    reactor B extends A{}
+                """.formatted(target));
+
+            if(target.supportsInheritance()) {
```
Comment (lhstrh) [inline org.lflang.tests/src/org/lflang/tests/compiler/LinguaFrancaValidationTest.java:824]:
```suggestion
            if (target.supportsInheritance()) {
```

File: org.lflang.tests/src/org/lflang/tests/compiler/LinguaFrancaValidationTest.java [hunk 3]
```diff
@@ -812,6 +812,41 @@ public void testPreambleVisibility() throws Exception {
...
+
+    @Test
+    public void testFederationSupport() throws Exception {
+        for (Target target : Target.values()) {
+            Model model = parseWithoutError("""
+                    target %s
+                    federated reactor {}
+                """.formatted(target));
+
+            if(target.supportsFederated()) {
```
Comment (lhstrh) [inline org.lflang.tests/src/org/lflang/tests/compiler/LinguaFrancaValidationTest.java:841]:
```suggestion
            if (target.supportsFederated()) {
```

File: org.lflang/src/org/lflang/validation/LFValidator.java [hunk 4]
```diff
@@ -969,6 +984,11 @@ public void checkReactor(Reactor reactor) throws IOException {
         }
 
         if (reactor.isFederated()) {
+            if(!target.supportsFederated()) {
```
Comment (lhstrh) [inline org.lflang/src/org/lflang/validation/LFValidator.java:987]:
```suggestion
            if (!target.supportsFederated()) {
```

**Review by lhstrh — APPROVED** [review]
Looks great! Thanks for adding the tests also.

**Comment by cmnrd** [comment]
Looks like my refactoring also revealed some other issues in the tests. This should be fixed now.

**Pull request merged**